### PR TITLE
Code improvements related to performance and style

### PR DESCRIPTION
**Pull request by NelsonVides**
For more info you can check the commits one by one, they all touch different topics and have descriptions in their messages.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7890.1](https://travis-ci.org/esl/MongooseIM/jobs/646464460) / Erlang 22.0 / small_tests / fe3b88073cd0780567877338dd2ac1d3a17e836b
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2621/7890/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2621/7890/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-02-05_15.38.24/lib.mongooseim.logs/run.2020-02-05_15.38.25/suite.log.html)

File: src/mod_ping.erl [hunk 1]
```diff
@@ -56,12 +56,12 @@
          user_keep_alive/2,
          user_ping_response/4]).
 
--record(state, {host = <<"">>,
+-record(state, {host = jid:lserver(),
```
Comment (michalwski) [inline src/mod_ping.erl]: Here the default value of the field is defined. `jid:lserver()` is a type so creation of the record will cause a undef exception when `host` is not specified.

**Comment by michalwski** [comment]
Oh, I see that benchmark results are in commit messages. That's enough, thanks :)

**Comment by mongoose-im** [comment]
[7900.1](https://travis-ci.org/esl/MongooseIM/jobs/648000224) / Erlang 22.0 / small_tests / 8581be62289c961cae781cdcc7b03f0048213f97
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2621/7900/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2621/7900/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-02-09_11.50.39/lib.mongooseim.logs/run.2020-02-09_11.50.39/suite.log.html)

File: src/mod_ping.erl [hunk 2]
```diff
@@ -56,12 +56,12 @@
          user_keep_alive/2,
          user_ping_response/4]).
 
--record(state, {host = <<"">>,
+-record(state, {host = <<"">> :: jid:lserver(),
```
Comment (michalwski) [inline src/mod_ping.erl]: The type of the `state` is defined below. No need to add type definition here.

Comment (michalwski) [inline src/mod_ping.erl]:
```suggestion
-record(state, {host = <<"">>,
```

**Comment by mongoose-im** [comment]
[7920.1](https://travis-ci.org/esl/MongooseIM/jobs/648827313) / Erlang 22.0 / small_tests / 561d772ad4d0e53450b0d3979fe4f15df90aacbb
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2621/7920/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2621/7920/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-02-11_10.37.45/lib.mongooseim.logs/run.2020-02-11_10.37.45/suite.log.html)

**Comment by mongoose-im** [comment]
[7922.1](https://travis-ci.org/esl/MongooseIM/jobs/648845757) / Erlang 22.0 / small_tests / 8f45e8b26931f2929ee4d6629aa9427f1fd939a3
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2621/7922/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2621/7922/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-02-11_11.53.29/lib.mongooseim.logs/run.2020-02-11_11.53.30/suite.log.html)

**Pull request merged**